pkg/instance/handler.go

```diff
@@ -379,6 +380,64 @@ func (h Handler) Pause(c *gin.Context) {
 	c.Status(http.StatusAccepted)
 }
 
+// Reset instance
+func (h Handler) Reset(c *gin.Context) {
+	// swagger:route PUT /instances/{id}/reset resetInstance
+	//
+	// Reset instance
+	//
+	// Resetting an instance will completely destroy it and redeploy using the same parameters
+	//
+	// Security:
+	//	oauth2:
+	//
+	// responses:
+	//	202:
+	//	400: Error
+	//	401: Error
+	//	403: Error
+	//	404: Error
+	idParam := c.Param("id")
+	id, err := strconv.ParseUint(idParam, 10, 32)
+	if err != nil {
+		_ = c.AbortWithError(http.StatusBadRequest, fmt.Errorf("failed to parse id: %s", err))
+		return
+	}
+
+	token, err := handler.GetTokenFromHttpAuthHeader(c)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	user, err := handler.GetUserFromContext(c)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	instance, err := h.instanceService.FindById(uint(id))
```

Comment:
We call a findBy in reset as well. Can we move the ACL check CanWriteInstance into the service and save one of the findBy calls?

Reply:
I'm not sure if that belongs in the service but we can discuss this further. However I think it's out of the scope for this PR.